src/Authenticator/LoginHandler.php

```diff
@@ -232,7 +234,9 @@ public function finishRegistration(HTTPRequest $request): HTTPResponse
         $sessionMember = $store ? $store->getMember() : null;
         $loggedInMember = Security::getCurrentUser();
 
-        if (is_null($loggedInMember) && is_null($sessionMember)) {
+        if ((is_null($loggedInMember) && is_null($sessionMember))
+            || !$this->getSudoModeService()->check($request->getSession())
```

Comment:
We should probably cover "start" too. The only reason is that I know backup codes will create new backup codes on "start".

Reply:
I wondered about this, but figured that since it's an edge case to get to GET methods without sudo mode active that we could save lines by only implementing the checks on POST methods

Reply:
Without this you could probably circumvent needing sudo to reset your backup codes though?

Reply:
That sounds like it should be a POST action to me. Can you point me at the part of the code that does it?

Reply:
https://github.com/silverstripe/silverstripe-mfa/blob/ec7007d97ce89f4a8f289b9537979fd80bdb6de0/src/BackupCode/RegisterHandler.php#L52

We discussed this a little at the time. It's a bit odd, but we need the codes to be saved already as they're presented to the user.